compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/PromptAsCodeCodeModificationTask.java

```diff
@@ -282,75 +317,196 @@ private boolean isNotNPCallCall(FunctionCallExpressionNode functionCallExpressio
+        TemplateExpressionNode promptRawTemplate = NodeFactory.createTemplateExpressionNode(
+                SyntaxKind.RAW_TEMPLATE_EXPRESSION, null, BACKTICK, prompt, BACKTICK);
+
+        Optional<ExpressionNode> naturalModelArg = Optional.empty();
+
+        Optional<ParenthesizedArgList> parenthesizedArgListOptional = naturalExpressionNode.parenthesizedArgList();
+        if (parenthesizedArgListOptional.isPresent()) {
+            ParenthesizedArgList parenthesizedArgList = parenthesizedArgListOptional.get();
+            SeparatedNodeList<FunctionArgumentNode> arguments = parenthesizedArgList.arguments();
+            if (!arguments.isEmpty()) {
+                FunctionArgumentNode functionArgumentNode = arguments.get(0);
+                if (functionArgumentNode instanceof PositionalArgumentNode positionalArgumentNode) {
+                    // TODO: validation to be done first
+                    naturalModelArg = Optional.of(positionalArgumentNode.expression());
+                }
+            }
+        }
 
-        private final ModifierData modifierData;
+        SeparatedNodeList<FunctionArgumentNode> arguments =
+                naturalModelArg.isPresent() ?
+                        NodeFactory.createSeparatedNodeList(
+                                promptRawTemplate,
+                                COMMA,
+                                createContextMappingExpressionNode(naturalModelArg.get())
+                        ) :
+                        NodeFactory.createSeparatedNodeList(
+                                promptRawTemplate
+                        );
+        return NodeFactory.createFunctionCallExpressionNode(
+                createNPCallQualifiedNameReferenceNode(npPrefix),
+                OPEN_PAREN,
+                arguments,
+                CLOSE_PAREN
+        );
+    }
 
-        FunctionModifier(ModifierData modifierData) {
-            this.modifierData = modifierData;
+    private static NodeList<Node> getRawTemplateContent(NaturalExpressionNode naturalExpressionNode,
+                                                        SemanticModel semanticModel) {
+        NodeList<Node> prompt = naturalExpressionNode.prompt();
+        List<Node> modifiedNodes = new ArrayList<>();
+        boolean modified = false;
+        MinutiaeList emptyMinutiaeList = AbstractNodeFactory.createEmptyMinutiaeList();
+
+        boolean hasInsertions = false;
+        for (Node node : prompt) {
+            if (!(node instanceof LiteralValueToken literalValueToken)) {
+                modifiedNodes.add(node);
+                hasInsertions = true;
+                continue;
+            }
+
+            SyntaxKind kind = literalValueToken.kind();
+            String text = literalValueToken.text();
+
+            if (text.isEmpty()) {
+                modifiedNodes.add(node);
+                continue;
+            }
+
+            String updatedText = text.replace("\\}", "}");
```

Comment:
Can you tell a example prompt that we need this

Reply:
See https://github.com/ballerina-platform/module-ballerina-np/pull/7/commits/e5d594a9be34a5a00323e3beedef6c78eb6203db#diff-1b5b7dfdee4f85882ce2fdb5043b016d1ee83beddba14d9e66185498faec633aR99